wezterm-gui/src/termwindow/mod.rs

```diff
@@ -2507,6 +2507,28 @@ impl TermWindow {
             ToggleFullScreen => {
                 self.window.as_ref().unwrap().toggle_fullscreen();
             }
+            #[cfg(target_os = "macos")]
+            ToggleFloatingWindow => {
+                let window = self.window.clone().unwrap();
+
+                promise::spawn::spawn(async move {
+                    match window.level().await {
```

Comment:
without a `level()` function to retrive the current window level how can we perform the toggle?

Reply:
I'm not opposed to adding a way to understand the level. Having a method that returns a Future makes it a bit awkward to do that in certain contexts.

Maybe adding `ALWAYS_ON_TOP` and `ALWAYS_ON_BOTTOM` to `WindowState` would satisfy this?

https://github.com/wez/wezterm/blob/f07cc837166dc5f29df56afb42af3d6887ec6a33/window/src/lib.rs#L102-L114

WindowState is exposed to lua via [window:get_dimensions](https://wezfurlong.org/wezterm/config/lua/window/get_dimensions.html)

https://github.com/wez/wezterm/blob/f07cc837166dc5f29df56afb42af3d6887ec6a33/wezterm-gui/src/scripting/guiwin.rs#L115-L131

That said, this is already another async method call and already awkward. I suppose the advantage is that we wouldn't need to make multiple different async calls to resolve more information about the window overall.

Reply:
looks nice, yeah async method is awkward, I agree with you. I'll try adding this to the state!